.github/workflows/release-publish.yml

```diff
@@ -51,25 +48,97 @@ jobs:
+        uses: docker/login-action@v3.0.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v3.0.0
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Build
+        uses: docker/build-push-action@v5.1.0
+        with:
+          context: ./docker/images/n8n
+          build-args: |
+            N8N_VERSION=${{ needs.publish-to-npm.outputs.release }}
+          platforms: linux/amd64,linux/arm64
+          provenance: false
+          push: true
+          tags: |
+            ${{ secrets.DOCKER_USERNAME }}/n8n:${{ needs.publish-to-npm.outputs.release }}
+            ghcr.io/${{ github.repository_owner }}/n8n:${{ needs.publish-to-npm.outputs.release }}
+
+  create-github-release:
+    name: Create a GitHub Release
+    needs: [publish-to-npm, publish-to-docker-hub]
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    timeout-minutes: 5
+
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
       - name: Create a Release on GitHub
         uses: ncipollo/release-action@v1
         with:
           commit: ${{github.event.pull_request.base.ref}}
-          tag: 'n8n@${{env.RELEASE}}'
+          tag: 'n8n@${{ needs.publish-to-npm.outputs.release }}'
           prerelease: true
           makeLatest: false
           body: ${{github.event.pull_request.body}}
 
+  trigger-release-note:
+    name: Trigger a release note
+    needs: [publish-to-npm, create-github-release]
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
       - name: Trigger a release note
-        continue-on-error: true
-        run: curl -u docsWorkflows:${{ secrets.N8N_WEBHOOK_DOCS_PASSWORD }} --request GET 'https://internal.users.n8n.cloud/webhook/trigger-release-note' --header 'Content-Type:application/json' --data '{"version":"${{env.RELEASE}}"}'
-
-      # - name: Merge Release into 'master'
-      #   run: |
-      #     git fetch origin
-      #     git checkout --track origin/master
-      #     git config user.name "Jan Oberhauser"
-      #     git config user.email jan.oberhauser@gmail.com
-      #     git merge --ff n8n@${{env.RELEASE}}
-      #     git push origin master
-      #     git push origin :${{github.event.pull_request.base.ref}}
+        run: curl -u docsWorkflows:${{ secrets.N8N_WEBHOOK_DOCS_PASSWORD }} --request GET 'https://internal.users.n8n.cloud/webhook/trigger-release-note' --header 'Content-Type:application/json' --data '{"version":"${{ needs.publish-to-npm.outputs.release }}"}'
+
+  merge-back-into-master:
+    name: Merge back into master
+    needs: [publish-to-npm, create-github-release]
+    if: ${{ github.event.pull_request.merged == true && !contains(github.event.pull_request.labels.*.name, 'release:patch') }}
```

Comment:
Isn't there also label `release:major`? Should those be merged back as well? The PR description says:

> and if the release was labeled with release:minor, the newly created tab is automatically merged back into master.

Reply:
yeah, this will work for `release:major` as well. I just didn't want to document that bit yet, until we have first finalized the updated release process for major releases.